### DISCUSSION TITLE
[@types/lodash] Lodash fp getOr function type is broken

### DISCUSSION THREAD

**Discussion by StefanoA1** [DefinitelyTyped/DefinitelyTyped#62072] [Issues with a @types package] [answered]
The` const foo` is still possibly null although the a default value is provided, the type should be string (the same as the default value) - using lodash/fp:

```
type Foo = string | null;
type SuperFoo = {
  data?: Foo;
};

const state: SuperFoo = {
  data: 'hey'
};

const foo = getOr('', ['data'], state);// const foo is still possibly undefined although the a default value is provided, the type should be Foo or string (default value)
```

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
This is the correct behavior. https://lodash.com/docs/4.17.15#get:

> If the resolved value is undefined, the defaultValue is returned in its place.

If the value is null, `get()` will return null.

**Reply by aj-r** [reply]
I'm not sure what you mean - `string | (string | null)` is the same as `string | null`.

**Reply by StefanoA1** [reply]
sorry I meant `undefined | (string | null)` I updated the topic, but it was finally a TS issue, I updated TS and it seems to work.
